Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -8399,12 +8399,15 @@ bool WebPage::isUsingUISideCompositing() const
 }
 
 #if ENABLE(NETWORK_CONNECTION_INTEGRITY)
-void WebPage::setLookalikeCharacterStrings(Vector<String>&& strings)
+void WebPage::setLookalikeCharacterStrings(Vector<WebCore::LookalikeCharactersSanitizationData>&& strings)
 {
     m_lookalikeCharacterStrings.clear();
     m_lookalikeCharacterStrings.reserveInitialCapacity(strings.size());
-    for (auto& string : strings)
-        m_lookalikeCharacterStrings.add(string);
+    for (auto& data : strings) {
```

Comment:
We should also double check that `data.domain` always represents a valid hash key here, to avoid hash table corruption (e.g. by trying to add a null string which represents the empty value).